client/src/Input.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+
+function Input({ onAddVideo }) {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+
+  const handleSubmit = (e) => {
+  e.preventDefault();
+  if (isValidYouTubeURL(url) && isValidTitle(title)) {
+    const embedUrl = generateEmbedURL(url);
+    onAddVideo(title, embedUrl);
+    setTitle("");
+    setUrl("");
+  } else {
+    alert("Please enter a valid YouTube URL and a non-empty title.");
+  }
+  };
+  
+  const isValidTitle = (title) => {
```

Comment:
It's good User Experience that you're validating in the front end app rather than sending the request to the back end and only allowing that to validate. This way, the user gets feedback quicker, without waiting for the request to go to the server and come back. In a production app, you'd probably validate on the server as well just for security purposes, in case someone sent a request directly to it.